### PR TITLE
[Build] Dropped Bazel 5.x and Upgraded Bazel 6.4.0

### PR DESCRIPTION
**Pull request by veblush**
- Dropped Bazel 5.x support aligned with [Foundational C++ support policy: Bazel](https://opensource.google/documentation/policies/cplusplus-support).
- Upgraded Bazel 6 to 6.4.0 (the latest)
- Updated gRPC Bazel documentation to be aligned with the policy above.

### CODE REVIEW

File: templates/doc/bazel_support.md.template [hunk 1]
```diff
@@ -36,8 +36,12 @@
   
   ${ '##' } Supported Versions
   
-  In general, gRPC supports building with the latest patch release of the two most
-  recent LTS versions of Bazel. However individual releases may have a broader
+  gRPC supports building with the latest stable release of Bazel,
+  as well as the previous major version release for at least 6 months
+  after it transitions into maintenance mode.
+  This is consistent with the supported build systems of
+  [the Foundational C++ Support Policy](https://opensource.google/documentation/policies/cplusplus-support).
```
Comment (gnossen) [inline templates/doc/bazel_support.md.template:43]: NIt: The _Google_ Foundational C++ Support Policy. We're a CNCF project.

**Review by gnossen — APPROVED** [review]
LGTM pending grpc-io email.

File: templates/doc/bazel_support.md.template (hunk 1 above)
Comment (veblush) [reply templates/doc/bazel_support.md.template]: Thanks! Done.

**Comment by veblush** [comment]
cc @acozzette

**Pull request merged**